diff --git a/include/linux/ipc_namespace.h b/include/linux/ipc_namespace.h
--- a/include/linux/ipc_namespace.h
+++ b/include/linux/ipc_namespace.h
@@ -11,6 +11,7 @@
 #include <linux/refcount.h>
 #include <linux/rhashtable-types.h>
+#include <linux/android_kabi.h>
 struct user_namespace;
 
 struct ipc_ids {
@@ -78,7 +79,13 @@ struct ipc_namespace {
 	struct llist_node mnt_llist;
 
 	struct ns_common ns;
+	ANDROID_KABI_RESERVE(1);
+	ANDROID_KABI_RESERVE(2);
+	ANDROID_KABI_RESERVE(3);
+	ANDROID_KABI_RESERVE(4);
+	ANDROID_KABI_RESERVE(5);
+	ANDROID_KABI_RESERVE(6);
 } __randomize_layout;
 
 extern struct ipc_namespace init_ipc_ns;