diff --git a/kernel/module.c b/kernel/module.c
index bcadcef319dc..3794f311ba02 100644
--- a/kernel/module.c
+++ b/kernel/module.c
@@ -1289,7 +1289,7 @@ static int check_version(const struct load_info *info,
 bad_version:
 	pr_warn("%s: disagrees about version of symbol %s\n",
 	       info->name, symname);
-	return 0;
+	return 1;
 }
 
 static inline int check_modstruct_version(const struct load_info *info,
