diff --git a/include/linux/netdevice.h b/include/linux/netdevice.h
index cc9d2358c35f..f2a55806cc0e 100644
--- a/include/linux/netdevice.h
+++ b/include/linux/netdevice.h
@@ -2078,7 +2078,7 @@ struct net_device {
 #endif
 	const struct ethtool_ops *ethtool_ops;
 #ifdef CONFIG_NET_L3_MASTER_DEV
-	const struct l3mdev_ops	*l3mdev_ops;
+//	const struct l3mdev_ops	*l3mdev_ops;
 #endif
 #if IS_ENABLED(CONFIG_IPV6)
 	const struct ndisc_ops *ndisc_ops;
@@ -2322,7 +2322,11 @@ struct net_device {
 	ANDROID_KABI_RESERVE(5);
 	ANDROID_KABI_RESERVE(6);
 	ANDROID_KABI_RESERVE(7);
-	ANDROID_KABI_RESERVE(8);
+#if defined(CONFIG_NET_L3_MASTER_DEV)
+	ANDROID_KABI_USE(8, const struct l3mdev_ops	*l3mdev_ops;);
+#else
+  	ANDROID_KABI_RESERVE(8);
+#endif
 };
 #define to_net_dev(d) container_of(d, struct net_device, dev)
 
